run.rb

```diff
@@ -0,0 +1,61 @@
+require_relative './app'
+require_relative './book'
+
+class Run
+  include App
+  include RentalListing
+  include BookLisitng
+  include PeopleListing
+
+  def initialize
+    @books = []
+    @peoples = []
+    @rentals = []
+  end
+
+  def main
+    loop do
+      puts "\n\nWelcome to School Library APP! \n\n"
+      print 'Please Choose an option by enerin a number: '
+      puts "
+        1 - List all books
+        2 - List all people
+        3 - Create a person
+        4 - Create a book
+        5 - Create a rental
+        6 - List all rentals for a given person id
+        7 - Exit
+        "
+      print 'option: '
+      option = gets.chomp
+      show(option)
+    end
+  end
+
+  def show(option)
+    case option
+    when '1'
+      list_all_books(@books)
+    when '2'
+      list_all_peoples(@peoples)
+    when '6'
+      list_rentals(@rentals, @peoples)
+    when '7'
```

Comment:
- [ ] In the `list_rentals` method, kindly put all, codes responsible for **_taking the user's input and validating the user's inputs_** should be in a separate method, module or class while the list_rental method should be responsible for listing the rental instances as we have it in the `list_all_books` and `list_all_peoples` methods. Kindly implement this change. 👍 💻

Reply:
@Gambit142  Nice observation, I will refactor these methods to validate inputs separately modules